architecture/2025-04-compute-platform-extensibility.md

```diff
@@ -0,0 +1,505 @@
+1. Application Definition and Deployment: The application definition files (app.bicep) will be extended to include ACI-specific properties and configurations. The Radius deployment engine will be enhanced to interpret these properties and deploy applications to ACI environments. This includes creating and managing ACI container groups, configuring networking, and handling ACI-specific resource settings.
+1. Container Definitions: The container definitions within the application definition files will allow users to deploy their containers to both Kubernetes and ACI without significant changes.
+1. Extension Properties, aka "punch-through": The Radius codebase will be enhanced to support extension properties in the application definition files. These properties will allow users to apply ACI-specific configurations to their containers, such as custom resource settings, networking configurations, and other ACI-specific parameters.
+1. Error Handling and Logging: The error handling and logging mechanisms within the Radius codebase will be updated to account for ACI-specific scenarios. This includes handling errors related to ACI resource creation, configuration, and deployment, as well as providing detailed logs for troubleshooting and diagnostics.
+1. Gateways: Radius will provision the necessary Azure resources to establish gateways for routing internet traffic to services hosted in ACI. This includes defining and creating gateways and secret stores for use in ACI containers.
+1. TBD: Secret Management: Radius will provision an Azure KeyVault resource on behalf of the user to store and manage secrets. These secrets can then be referenced in the application definition files to inject secret values as environment variables in ACI containers or to authenticate into services.
+
+<!--
+### Architecture Diagram
+Provide a diagram of the system architecture, illustrating how different
+components interact with each other in the context of this proposal.
+
+Include separate high level architecture diagram and component specific diagrams, wherever appropriate.
+-->
+
+### Detailed Design
+
+#### Key Components and Relationships
+
+1. Azure Container Instances (ACI) Integration
+    * `corerp/renderers/aci` implements ACI-specific resource rendering logic
+    * `corerp/renderers/aci/gateway` provides gateway functionality for ACI
+    * `corerp/renderers/aci/manualscale` handles scaling for ACI containers
+1. Renderer Multiplexer
+    * `corerp/renderers/mux` routes requests to appropriate renderers based on compute environment
+    * Selects between Kubernetes and ACI renderers based on environment configuration
+
+This architecture supports multi-cloud deployments with a unified programming model, where applications can be defined once and deployed to either Kubernetes or Azure Container Instances environments based on the environment configuration.
+
+#### Understanding the Radius/ACI Architecture Relationships
```

Comment:
Any changes needed in the deployment processor?

Reply:
Yes - the deployment processor requires a resource group property to be set for ACI. I will update the doc.